website/content/en/docs/building-operators/golang/webhook.md

```diff
@@ -0,0 +1,300 @@
+but as an example, we can reinforce that the default value for `spec.size` should be `3`, by adding the following logic to 
+the `Default` function (note that this is already handled by the CRD defaulting and is technically completely superfluous):
+
+```sh
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *Memcached) Default() {
+	memcachedlog.Info("default", "name", r.Name)
+
+	if r.Spec.Size == 0 {
+		r.Spec.Size = 3
+	}
+}
+```
+
+For validation, we can enforce that the size of the cluster follows a rule that is difficult or impossible 
+to describe with OpenAPI, for example, that the size of the cluster always remain an odd number.
+
+In order to perform this validation, we can simply add below `validateOdd` function that will perform this check.
+
+```sh
+func validateOdd(n int32) error {
+	if n%2 == 0 {
+		return errors.New("Cluster size must be an odd number")
+	}
+	return nil
+}
+```
+
+At the end, call this method from `ValidateCreate` and `ValidateUpdate` function as shown below.
+
+```sh
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *Memcached) ValidateCreate() error {
+	memcachedlog.Info("validate create", "name", r.Name)
+	return validateOdd(r.Spec.Size)
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *Memcached) ValidateUpdate(old runtime.Object) error {
+	memcachedlog.Info("validate update", "name", r.Name)
+	return validateOdd(r.Spec.Size)
+}
+```
+
+This function gets called whenever there is an object deletion happens.
+
+```sh
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (r *Memcached) ValidateDelete() error {
+	memcachedlog.Info("validate delete", "name", r.Name)
+
+	// TODO(user): fill in your validation logic upon object deletion.
+	return nil
+}
+```
+
+### Generate webhook manifests and enable webhook deployment 
+
+Once your webhooks are implemented, all that’s left is to create the WebhookConfiguration manifests required to register your webhooks with Kubernetes:
+
+```sh
+$ make manifests
+```
+
+You will need to enable cert-manager and webhook deployment in order to deploy these webhooks properly. 
+To do so, edit the `config/default/kustomize.yaml` and uncomment the sections marked by `[WEBHOOK]` and `[CERTMANAGER]` comments. 
+More detail on this step can be found in the [Kubebuilder documentation](https://book.kubebuilder.io/cronjob-tutorial/running-webhook.html).
+
+### Update main.go so that running locally works 
+
+To ensure that operator running locally continues working, ensure that there is a check that prevents the webhooks from being started when the `ENABLE_WEBHOOKS` flag is set to `false`. To do so, edit `main.go`, and add the following check around the call to `SetupWebhookWithManager` if it’s not already present:
+
+```sh
+if os.Getenv("ENABLE_WEBHOOKS") != "false" {
```

Comment:
I'd say lets direct users to refer to the upstream documentation for running webhooks locally. Its already explained here (https://book.kubebuilder.io/cronjob-tutorial/main-revisited.html?highlight=enable_webhooks#you-said-something-about-main). This seems to be like the same repetition.

Reply:
Modified.